messenger: crash after login while getting list of chats

# Bug Report
App crashes shortly after login while trying to get the chats; likely a result of this recent [getChats async refactor PR](https://github.com/status-im/status-desktop/pull/9363)

Crash backtrace:
```
Traceback (most recent call last)
/home/ltinkl/git/status/status-desktop/src/nim_status_client.nim(180) nim_status_client
/home/ltinkl/git/status/status-desktop/src/nim_status_client.nim(177) mainProc
/home/ltinkl/git/status/status-desktop/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/home/ltinkl/git/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/home/ltinkl/git/status/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/home/ltinkl/git/status/status-desktop/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(566) onSlotCalled
/home/ltinkl/git/status/status-desktop/src/app/core/signals/signals_manager.nim(56) receiveSignal
/home/ltinkl/git/status/status-desktop/src/app/core/signals/signals_manager.nim(53) processSignal
/home/ltinkl/git/status/status-desktop/src/app/core/eventemitter.nim(57) emit
/home/ltinkl/git/status/status-desktop/src/app_service/service/contacts/service.nim(165) :anonymous
/home/ltinkl/git/status/status-desktop/src/app_service/service/contacts/service.nim(159) updateAndEmitStatuses
/home/ltinkl/git/status/status-desktop/src/app/core/eventemitter.nim(57) emit
/home/ltinkl/git/status/status-desktop/src/app/modules/main/controller.nim(218) :anonymous
/home/ltinkl/git/status/status-desktop/src/app/modules/main/io_interface.nim(135) contactsStatusUpdated
/home/ltinkl/git/status/status-desktop/src/app/modules/main/module.nim(840) contactsStatusUpdated
/home/ltinkl/git/status/status-desktop/src/app/modules/shared_models/active_section.nim(233) setOnlineStatusForMember
/home/ltinkl/git/status/status-desktop/src/app/modules/shared_models/section_item.nim(265) setOnlineStatusForMember
/home/ltinkl/git/status/status-desktop/src/app/modules/shared_models/member_model.nim(281) setOnlineStatus
/home/ltinkl/git/status/status-desktop/src/app/modules/shared_models/member_model.nim(154) findIndexForMember
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

The crash happens here:
```nim
  proc findIndexForMember(self: Model, pubKey: string): int =
>>>    for i in 0 ..< self.items.len: <<<
      if(self.items[i].pubKey == pubKey):
        return i

    return -1
```

Before the crash, I see a lot of repeated calls like this:
```
WRN 2023-02-09 15:07:18.460+01:00 trying to get chat data for an unexisting chat id topics="chat-service" tid=213485 file=service.nim:315 chatId=0x03073514d4c14a7d10ae9fc9b0f05abc904d84166a6ac80add58bf6a3542a4e50a865bf58e-7298-4c70-b426-ebe8e3edf24d
ERR 2023-02-09 15:07:18.460+01:00 error: requested community doesn't exists  topics="community-service" tid=213485 file=service.nim:559 communityId=
WRN 2023-02-09 15:07:18.462+01:00 trying to get chat data for an unexisting chat id topics="chat-service" tid=213485 file=service.nim:315 chatId=0913691f-a685-4a41-bb4b-44c472d60889-0x04f2e7b3394dcbf0d03bba954dbedc0bd561951a8c31a00320c6b99a40145e655da6e689d19ced6a314c6dde31ce96feb0166f19472c2a1edbeeb939e3282bce14
ERR 2023-02-09 15:07:18.463+01:00 error: requested community doesn't exists  topics="community-service" tid=213485 file=service.nim:559 communityId=
```
... and so on

### Additional Information

- Status desktop version: master (f5d6c538c2918faecb3affd6a1b264f4efe7f795)
- Operating System: linux

Not sure if it is the same bug, but I am also not able to login. It just crashes after few seconds once password is entered

```
[~]-> status
INFO [02-13|19:48:40.565] Status backend initialized               backend=geth version=0.125.0 commit=1f9c1ceef IpfsGatewayURL=https://ipfs.status.im/
INF 2023-02-13 19:48:40.811+05:30 Version: 0.9.0                             topics="status-app" tid=18021 file=nim_status_client.nim:169
INF 2023-02-13 19:48:40.812+05:30 Commit: ef27bb027                          topics="status-app" tid=18021 file=nim_status_client.nim:170
INF 2023-02-13 19:48:40.812+05:30 Current date:                              topics="status-app" tid=18021 file=nim_status_client.nim:171 currentDateTime=2023-02-13T19:48:40+05:30
INF 2023-02-13 19:48:40.812+05:30 starting application controller...         topics="status-app" tid=18021 file=nim_status_client.nim:173
Debug: Network Cache Dir:  "/home/parvesh_monu/.config/Status/tmp/netcache" (:0, )
INF 2023-02-13 19:48:45.123+05:30 starting application...                    topics="status-app" tid=18021 file=nim_status_client.nim:176
WRN 2023-02-13 19:48:50.851+05:30 Error decoding signal                      topics="signals-manager" tid=18021 file=signals_manager.nim:43 err="Unknown signal received: mediaserver.started"
INF 2023-02-13 19:48:54.315+05:30 fetch mailservers                          topics="mailservers-service" tid=18021 file=service.nim:208 rpc_proc=mailservers_getMailservers response="(jsonrpc: \"2.0\", result: null, id: 0, error: nil)"
panic: runtime error: slice bounds out of range [:2] with capacity 0

goroutine 499 [running]:
github.com/waku-org/go-waku/waku/v2/utils.Multiaddress(0x1c002616360)
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/utils/enr.go:184 +0x5db
github.com/waku-org/go-waku/waku/v2/utils.EnodeToPeerInfo(0x7fe405620a20?)
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/utils/enr.go:207 +0x19
github.com/waku-org/go-waku/waku/v2/discv5.evaluateNode(0x1c002616360)
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/discv5/discover.go:232 +0x3f
github.com/ethereum/go-ethereum/p2p/enode.(*filterIter).Next(0x1c005387fe0)
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/ethereum/go-ethereum/p2p/enode/iter.go:119 +0x54
github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange.(*WakuPeerExchange).iterate(0x1c0002ed480, {0x7fe4071569a8, 0x1c00371fd00})
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange/waku_peer_exchange.go:327 +0x245
github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange.(*WakuPeerExchange).runPeerExchangeDiscv5Loop(0x1c0002ed480, {0x7fe4071569a8, 0x1c00371fd00})
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange/waku_peer_exchange.go:370 +0x28f
created by github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange.(*WakuPeerExchange).Start
	/home/jenkins/workspace/status-desktop/platforms/linux/vendor/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/peer_exchange/waku_peer_exchange.go:94 +0x1c5
Segmentation fault (core dumped)
```

--- a/src/app/global/app_sections_config.nim
+++ b/src/app/global/app_sections_config.nim
@@ -1,6 +1,8 @@
 const CHAT_SECTION_NAME* = "Messages"
 const CHAT_SECTION_ICON* = "chat"
 
+const LOADING_SECTION_ID* = "loadingSection"
+
 const COMMUNITIESPORTAL_SECTION_ID* = "communitiesPortal"
 const COMMUNITIESPORTAL_SECTION_NAME* = "Communities Portal"
 const COMMUNITIESPORTAL_SECTION_ICON* = "communities"

--- a/src/app/modules/main/controller.nim
+++ b/src/app/modules/main/controller.nim
@@ -46,7 +46,7 @@ type
     authenticateUserFlowRequestedBy: string
 
 # Forward declaration
-proc setActiveSection*(self: Controller, sectionId: string)
+proc setActiveSection*(self: Controller, sectionId: string, skipSavingInSettings: bool = false)
 
 proc newController*(delegate: io_interface.AccessInterface,
   events: EventEmitter,
@@ -320,10 +320,11 @@ proc getChannelGroups*(self: Controller): seq[ChannelGroupDto] =
 proc getActiveSectionId*(self: Controller): string =
   result = self.activeSectionId
 
-proc setActiveSection*(self: Controller, sectionId: string) =
+proc setActiveSection*(self: Controller, sectionId: string, skipSavingInSettings: bool = false) =
   self.activeSectionId = sectionId
-  let sectionIdToSave = if (sectionId == conf.SETTINGS_SECTION_ID): "" else: sectionId
-  singletonInstance.localAccountSensitiveSettings.setActiveSection(sectionIdToSave)
+  if not skipSavingInSettings:
+    let sectionIdToSave = if (sectionId == conf.SETTINGS_SECTION_ID): "" else: sectionId
+    singletonInstance.localAccountSensitiveSettings.setActiveSection(sectionIdToSave)
   self.delegate.activeSectionSet(self.activeSectionId)
 
 proc getNumOfNotificaitonsForChat*(self: Controller): tuple[unviewed:int, mentions:int] =

--- a/src/app/modules/main/io_interface.nim
+++ b/src/app/modules/main/io_interface.nim
@@ -177,7 +177,7 @@ method onNetworkDisconnected*(self: AccessInterface) {.base.} =
 method viewDidLoad*(self: AccessInterface) {.base.} =
   raise newException(ValueError, "No implementation available")
 
-method setActiveSection*(self: AccessInterface, item: SectionItem) {.base.} =
+method setActiveSection*(self: AccessInterface, item: SectionItem, skipSavingInSettings: bool = false) {.base.} =
   raise newException(ValueError, "No implementation available")
 
 method setActiveSectionById*(self: AccessInterface, id: string) {.base.} =

--- a/src/app/modules/main/module.nim
+++ b/src/app/modules/main/module.nim
@@ -356,92 +356,109 @@ method load*[T](
     activeSectionId = singletonInstance.userProfile.getPubKey()
 
   # Communities Portal Section
-  let communitiesPortalSectionItem = initItem(conf.COMMUNITIESPORTAL_SECTION_ID, SectionType.CommunitiesPortal, conf.COMMUNITIESPORTAL_SECTION_NAME,
-  amISectionAdmin = false,
-  description = "",
-  image = "",
-  icon = conf.COMMUNITIESPORTAL_SECTION_ICON,
-  color = "",
-  hasNotification = false,
-  notificationsCount = 0,
-  active = false,
-  enabled = true)
+  let communitiesPortalSectionItem = initItem(
+    conf.COMMUNITIESPORTAL_SECTION_ID,
+    SectionType.CommunitiesPortal,
+    conf.COMMUNITIESPORTAL_SECTION_NAME,
+    amISectionAdmin = false,
+    description = "",
+    image = "",
+    icon = conf.COMMUNITIESPORTAL_SECTION_ICON,
+    color = "",
+    hasNotification = false,
+    notificationsCount = 0,
+    active = false,
+    enabled = true,
+  )
   self.view.model().addItem(communitiesPortalSectionItem)
   if(activeSectionId == communitiesPortalSectionItem.id):
     activeSection = communitiesPortalSectionItem
 
   # Wallet Section
-  let walletSectionItem = initItem(conf.WALLET_SECTION_ID, SectionType.Wallet, conf.WALLET_SECTION_NAME,
-  amISectionAdmin = false,
-  description = "",
-  introMessage = "",
-  outroMessage = "",
-  image = "",
-  icon = conf.WALLET_SECTION_ICON,
-  color = "",
-  hasNotification = false,
-  notificationsCount = 0,
-  active = false,
-  enabled = singletonInstance.localAccountSensitiveSettings.getIsWalletEnabled())
+  let walletSectionItem = initItem(
+    conf.WALLET_SECTION_ID,
+    SectionType.Wallet,
+    conf.WALLET_SECTION_NAME,
+    amISectionAdmin = false,
+    description = "",
+    introMessage = "",
+    outroMessage = "",
+    image = "",
+    icon = conf.WALLET_SECTION_ICON,
+    color = "",
+    hasNotification = false,
+    notificationsCount = 0,
+    active = false,
+    enabled = singletonInstance.localAccountSensitiveSettings.getIsWalletEnabled(),
+  )
   self.view.model().addItem(walletSectionItem)
   if(activeSectionId == walletSectionItem.id):
     activeSection = walletSectionItem
 
   # Browser Section
-  let browserSectionItem = initItem(conf.BROWSER_SECTION_ID, SectionType.Browser, conf.BROWSER_SECTION_NAME,
-  amISectionAdmin = false,
-  description = "",
-  introMessage = "",
-  outroMessage = "",
-  image = "",
-  icon = conf.BROWSER_SECTION_ICON,
-  color = "",
-  hasNotification = false,
-  notificationsCount = 0,
-  active = false,
-  enabled = singletonInstance.localAccountSensitiveSettings.getIsBrowserEnabled())
+  let browserSectionItem = initItem(
+    conf.BROWSER_SECTION_ID,
+    SectionType.Browser,
+    conf.BROWSER_SECTION_NAME,
+    amISectionAdmin = false,
+    description = "",
+    introMessage = "",
+    outroMessage = "",
+    image = "",
+    icon = conf.BROWSER_SECTION_ICON,
+    color = "",
+    hasNotification = false,
+    notificationsCount = 0,
+    active = false,
+    enabled = singletonInstance.localAccountSensitiveSettings.getIsBrowserEnabled(),
+  )
   self.view.model().addItem(browserSectionItem)
   if(activeSectionId == browserSectionItem.id):
     activeSection = browserSectionItem
 
   # Node Management Section
-  let nodeManagementSectionItem = initItem(conf.NODEMANAGEMENT_SECTION_ID, SectionType.NodeManagement,
-  conf.NODEMANAGEMENT_SECTION_NAME,
-  amISectionAdmin = false,
-  description = "",
-  introMessage = "",
-  outroMessage = "",
-  image = "",
-  icon = conf.NODEMANAGEMENT_SECTION_ICON,
-  color = "",
-  hasNotification = false,
-  notificationsCount = 0,
-  active = false,
-  enabled = singletonInstance.localAccountSensitiveSettings.getNodeManagementEnabled())
+  let nodeManagementSectionItem = initItem(
+    conf.NODEMANAGEMENT_SECTION_ID,
+    SectionType.NodeManagement,
+    conf.NODEMANAGEMENT_SECTION_NAME,
+    amISectionAdmin = false,
+    description = "",
+    introMessage = "",
+    outroMessage = "",
+    image = "",
+    icon = conf.NODEMANAGEMENT_SECTION_ICON,
+    color = "",
+    hasNotification = false,
+    notificationsCount = 0,
+    active = false,
+    enabled = singletonInstance.localAccountSensitiveSettings.getNodeManagementEnabled(),
+  )
   self.view.model().addItem(nodeManagementSectionItem)
   if(activeSectionId == nodeManagementSectionItem.id):
     activeSection = nodeManagementSectionItem
 
   # Profile Section
-  let profileSettingsSectionItem = initItem(conf.SETTINGS_SECTION_ID, SectionType.ProfileSettings,
-  conf.SETTINGS_SECTION_NAME,
-  amISectionAdmin = false,
-  description = "",
-  introMessage = "",
-  outroMessage = "",
-  image = "",
-  icon = conf.SETTINGS_SECTION_ICON,
-  color = "",
-  hasNotification = self.calculateProfileSectionHasNotification(),
-  notificationsCount = 0,
-  active = false,
-  enabled = true)
+  let profileSettingsSectionItem = initItem(
+    conf.SETTINGS_SECTION_ID,
+    SectionType.ProfileSettings,
+    conf.SETTINGS_SECTION_NAME,
+    amISectionAdmin = false,
+    description = "",
+    introMessage = "",
+    outroMessage = "",
+    image = "",
+    icon = conf.SETTINGS_SECTION_ICON,
+    color = "",
+    hasNotification = self.calculateProfileSectionHasNotification(),
+    notificationsCount = 0,
+    active = false,
+    enabled = true,
+  )
   self.view.model().addItem(profileSettingsSectionItem)
   if(activeSectionId == profileSettingsSectionItem.id):
     activeSection = profileSettingsSectionItem
 
   self.browserSectionModule.load()
-  # self.nodeManagementSectionModule.load()
   self.profileSectionModule.load()
   self.stickersModule.load()
   self.networksModule.load()
@@ -451,12 +468,27 @@ method load*[T](
   self.nodeSectionModule.load()
   # Load wallet last as it triggers events that are listened by other modules
   self.walletSectionModule.load()
-  #self.communitiesPortalSectionModule.load()
 
   # Set active section on app start
-  # If section is profile then open chat by default
+  # If section is empty or profile then open the loading section until chats are loaded
   if activeSection.isEmpty() or activeSection.sectionType == SectionType.ProfileSettings:
-    self.setActiveSection(self.view.model().getItemBySectionType(SectionType.Chat))
+    # Set bogus Item as active until the chat is loaded
+    let loadingItem = initItem(
+      LOADING_SECTION_ID,
+      SectionType.LoadingSection,
+      name = "",
+      amISectionAdmin = false,
+      description = "",
+      image = "",
+      icon = "",
+      color = "",
+      hasNotification = false,
+      notificationsCount = 0,
+      active = false,
+      enabled = true,
+    )
+    self.view.model().addItem(loadingItem)
+    self.setActiveSection(loadingItem, skipSavingInSettings = true)
   else:
     self.setActiveSection(activeSection)
 
@@ -504,6 +536,9 @@ method onChatsLoaded*[T](
   # Set active section if it is one of the channel sections
   if not activeSection.isEmpty():
     self.setActiveSection(activeSection)
+
+  # Remove old loading section
+  self.view.model().removeItem(LOADING_SECTION_ID)
 
   self.view.chatsLoaded()
 
@@ -602,11 +637,11 @@ method setCommunityIdToSpectate*[T](self: Module[T], communityId: string) =
 method getActiveSectionId*[T](self: Module[T]): string =
   return self.controller.getActiveSectionId()
 
-method setActiveSection*[T](self: Module[T], item: SectionItem) =
+method setActiveSection*[T](self: Module[T], item: SectionItem, skipSavingInSettings: bool = false) =
   if(item.isEmpty()):
     echo "section is empty and cannot be made as active one"
     return
-  self.controller.setActiveSection(item.id)
+  self.controller.setActiveSection(item.id, skipSavingInSettings)
 
 method setActiveSectionById*[T](self: Module[T], id: string) =
     let item = self.view.model().getItemById(id)

--- a/src/app/modules/shared_models/section_item.nim
+++ b/src/app/modules/shared_models/section_item.nim
@@ -7,6 +7,7 @@ import ../../../app_service/common/types
 
 type
   SectionType* {.pure.} = enum
+    LoadingSection = -1
     Chat = 0
     Community,
     Wallet,

